### PR TITLE
Support job-level node_selector override in generated jobs

### LINKED ISSUE #2242: push-authentikos_test-infra_postsubmit Job stuck pending due to node selector
This has a node selector testing=test-pool which doesn't exist in the trusted cluster

### PR DESCRIPTION
**Pull request by clarketm**
fix #2242

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Review by fejta — APPROVED** [review]
Can we please configure the istio job generator to go to istio to review rather than the central prow team? 

This isn't prow functionality that needs our gating
/hold

**Comment by clarketm** [comment]
> Can we please configure the istio job generator to go to istio to review rather than the central prow team?
> 
> This isn't prow functionality that needs our gating
> /hold

The reason you are required for review in this case is because the *test-infra* team has ownership over jobs in `test-infra` job directory (which are being updated in this PR with new node selectors); `prow/config` which contains the generator logic is already owned by Istio T&R. If you don't mind relinquishing control over this directory to Istio I can delete that line from `CODEOWNERS`.

https://github.com/istio/test-infra/blob/4ec456594a03dffb65f8a1a5bca1d71b6e8a102c/CODEOWNERS#L10

**Pull request merged**
